File: Decode String.cpp
class Solution {
public:
    string decodeString(string s) {
        int i=0, count=0;
        string result="";
        string digit="";
        stack<int> nums;
        stack<string> words;
        while(i<s.length()){
            if(isdigit(s[i])){
               while(isdigit(s[i])){
                digit=digit+s[i];    
                i++; 
                }
               nums.push(stoi(digit));
                digit = "";
            }if(s[i]==']'){
                count--;
                string answer="";
                while(words.top()!="["){
                    answer = words.top() + answer;
                    words.pop();
                }words.pop();
                int k = nums.top();
                nums.pop();
                string one = answer;
                while(k>1){
                    answer = answer + one;
                    k--;
                }
                if(count==0 && words.empty())
                    result = result+answer;
                else
                    words.push(answer);
                i++;
        }
            else{
                if(s[i]=='[')
                    count++;
                string c="";
                c = c+s[i];
                words.push(c);
                i++;
            }
        }
        while(!nums.empty()){
            string answer="";
                while(words.top()!="["){
                    answer = words.top() + answer;
                    words.pop();
                }
                words.pop();
                int k = nums.top();
                nums.pop();
                string one= answer;
                while(k>1){
                    answer = answer + one;
                    k--;
                }
                if(count==0 && words.empty())
                    result = result+answer;
                else
                    words.push(answer);
        }   
            if(words.empty())
                return result;
             string answer = "";
             while(!words.empty()){
                 answer = words.top()+answer;
                 words.pop();
             }
        return result+answer;
    }
};

File: Decode String.py
class Solution:
    stack = []
    nums = []
    index = 0
    def decodeString(self, s: str) -> str:
        if self.index >= len(s):
            res = "".join(self.stack)
            self.stack.clear()
            self.nums.clear()
            return res
        if s[self.index].isdigit():
            temp = []
            while s[self.index].isdigit():
                temp.append(s[self.index])
                self.index += 1
            self.nums.append(int("".join(temp)))
            return self.decodeString(s)

        elif s[self.index] == "]":
            temp = []
            answer = []
            while self.stack[len(self.stack)-1] != "[":
                temp.append(self.stack.pop())
            self.stack.pop()
            temp.reverse()
            temp = "".join(temp)
            for i in range(self.nums.pop()):
                answer.append(temp)
            self.stack.append("".join(answer))
        else:
            self.stack.append(s[self.index])
        self.index += 1  
        return self.decodeString(s)

                
